youmubot-osu/src/discord/mod.rs

```diff
@@ -319,14 +321,36 @@ async fn add_user(
     user: models::User,
     data: &TypeMap,
 ) -> Result<()> {
+    let osu_client = data.get::<OsuClient>().unwrap();
+
+    let pp = [Mode::Std, Mode::Taiko, Mode::Catch, Mode::Mania]
+        .into_iter()
+        .map(|mode| async move {
+            osu_client
+                .user(UserID::ID(user.id), |f| f.mode(mode))
+                .await
+                .map(|u_opt| u_opt.map(|u| u.pp.unwrap_or(0.0)))
+        })
+        .collect::<stream::FuturesOrdered<_>>()
+        .try_collect::<Vec<_>>()
+        .await?;
+
+    let cache = data.get::<BeatmapMetaCache>().unwrap();
+
+    let scores = osu_client
+        .user_best(UserID::ID(user.id), |f| f.mode(Mode::Std).limit(100))
+        .await?;
+
+    let std_weight_map_length = calculate_weighted_map_length(&scores, cache, Mode::Std).await?;
```

Comment:
maybe do these in parallel: https://docs.rs/futures-util/latest/futures_util/macro.join.html